IPs/RSKIP178.md

```diff
@@ -0,0 +1,375 @@
+
+This RSKIP represents one of the changes to the RSK consensus that is required to achieve security under the SFAMM model.  
+
+
+
+
+## Motivation
+
+The only contribution to the cryptoeconomic security of an independent proof-of-work chain without subsidy is the existence of high transaction fees. In the case of sidechain 1:1 pegged to a primary chain that is protected by merge-mining, there is a high strategic alignment between the primary chain miners and merge-miners, so the honest majority and the fact that attacks to the secondary chain often become attributable are the two main assurances for the security for the sidechain. Nevertheless, RSK went further, and deployed the Armadillo system to provide fork-awareness. This subsystem creates a time window to trigger an emergency response by the community, in case of long reorg attempts.  Nevertheless, Armadillo requires a separate communication channel and the existence of at least one honest node monitoring the Bitcoin network. The SFAMM protocol and security model captures the benefits of Armadillo, makes Armadillo decentralized, and formalizes the protocol algorithms, assumptions and guarantees. The SFAMM protocol is ideal for RSK as it focuses not only on the hashrate capability of the attacker but also on the processing time that will be required for a successful attack.
+
+This RSKIP proposes a consensus change that enables RSK nodes to:
+
+- compute the active SHA256 hashrate and detect variations on the active hashrate that are compatible with an ongoing attack
+
+- Accumulate all the active SHA256 hashrate in the RSK blockchain, with independence of the amount of hashrate that is directly involved in merge-mining. This hashrate means more accumulated work, increasing the security against certain double-spend attacks. Also, the hashrate accumulated not only includes 100% of Bitcoin hashrate, but also hashrate from bitcoin forks.
+
+  
+
+
+## Specification
+
+ The RSK block header is a list of the following elements:
+
+```
+parentHash, unclesHash, coinbase,stateRoot, txTrieRoot, receiptTrieRoot, logsBloom, difficulty, number, gasLimit, gasUsed, timestamp, extraData, paidFees, minimumGasPrice, uncleCount, ummRoot [, mergeMiningFields ] 
+```
+
+The *mergeMiningFields* are the following:
+
+```
+bitcoinMergedMiningHeader,
+bitcoinMergedMiningMerkleProof,
+bitcoinMergedMiningCoinbaseTransaction
+```
+
+We define a **mergeMiningProof** to be a list containing the *mergeMiningFields*.
+
+The block in encoded as:
+
+```
+blockHeader, transactions, uncles
+```
+
+And *uncles* is a list of RSK block headers. We define a **heavyblock** to be a *mergeMiningProof* with certain properties that will be verified in consensus.
+
+We extend the list uncles to include, after all RSK block headers, a number of heavyblock elements (defined later). To maintain semantic coherence, the *uncles* field is renamed *difficultyContributors*. The field *uncleCount* is renamed as *difficultyContribution*.
```

Comment:
We extend the list of uncles to include, after all RSK block headers, a number of heavyblock elements (defined later). To maintain semantic coherence, the *uncles* field is renamed *difficultyContributors*. The field *uncleCount* is renamed as *difficultyContribution*.